orttraining/orttraining/python/training/execution_agent.py

```diff
@@ -107,33 +115,31 @@ def __init__(self, path_or_bytes, session_options=None, providers=None, provider
         self._training_agent = None
         self._inference_session = None
```

Comment:
can delete?

Reply:
I'd go further (and this is totally optional!) but I'd like it even better if we could remove this member altogether:

1. Pass `path_or_bytes, session_options, providers, provider_options` to the `C_TrainingAgent` initializer.
2. Create the `InferenceSession` as a private member of the C++ `TrainingAgent` class.

Reply:
I actually gave it a try and while it is not impossible but it is more effort than it seems and the benefits are not so much given this code is initialized once per training session, here is why, these options(path_or_bytes, session_options, providers, provider_options) are further normalized on python side in _create_inference_session and then some more pre-processing is done before the C++ code is called, hence for me to directly create inference session on C++ side I will need to replicated all this preprocessing in this file.

Reply:
I think you may have been trying to do something more complicated than I suggested.

My suggestion is not a blocker for this PR, but the use of `InferenceSession` in `TrainingAgent` should be a private implementation detail, since it doesn't need to be visible outside the `TrainingAgent` implementation.